examples/telegram/responses/pictures.py

```diff
@@ -0,0 +1,160 @@
+                    commands=["start", "restart"]
+                )
+            },
+        },
+    },
+    "pics": {
+        "ask_picture": {
+            RESPONSE: Response(text="Send me a picture"),
+            TRANSITIONS: {
+                ("pics", "send_one", 1.1): cnd.any(
+                    [
+                        messenger.cnd.message_handler(content_types=["photo"]),
+                        messenger.cnd.message_handler(
+                            func=doc_is_photo, content_types=["document"]
+                        ),
+                    ]
+                ),
+                ("pics", "send_many", 1.0): messenger.cnd.message_handler(
+                    content_types=["sticker"]
+                ),
+                ("pics", "repeat", 0.9): cnd.true(),
+            },
+        },
+        "send_one": {
+            # An HTTP path or a path to a local file can be used here.
+            RESPONSE: Response(text="Here's my picture!", image=Image(source=kitten_url)),
+            TRANSITIONS: {("root", "fallback"): cnd.true()},
+        },
+        "send_many": {
+            RESPONSE: Response(
+                text="Look at my pictures",
+                # An HTTP path or a path to a local file can be used here.
+                attachments=Attachments(files=[Image(source=kitten_url)] * 2),
+            ),
+            TRANSITIONS: {("root", "fallback"): cnd.true()},
+        },
+        "repeat": {
+            RESPONSE: "I cannot find the picture. Please, try again.",
+            TRANSITIONS: {
+                ("pics", "send_one", 1.1): cnd.any(
+                    [
+                        messenger.cnd.message_handler(content_types=["photo"]),
+                        messenger.cnd.message_handler(
+                            func=doc_is_photo, content_types=["document"]
+                        ),
+                    ]
+                ),
+                ("pics", "send_many", 1.0): messenger.cnd.message_handler(
+                    content_types=["sticker"]
+                ),
+                ("pics", "repeat", 0.9): cnd.true(),
+            },
+        },
+    },
+}
+
+
+def extract_data(ctx: Context, actor: Actor):  # A function to extract data with
+    message = ctx.framework_states[TELEGRAM_STATE_KEY].get("data")
+    if not message or (not message.photo and not doc_is_photo(message)):
+        return ctx
+    photo = message.document or message.photo[-1]
+    file = messenger.get_file(photo.file_id)
+    result = messenger.download_file(file.file_path)
+    with open("photo.jpg", "wb+") as new_file:
+        new_file.write(result)
+    return ctx
+
+
+interface = PollingTelegramInterface(messenger=messenger)
+
+
+# %%
+pipeline = Pipeline.from_script(
+    script=script,
+    start_label=("root", "start"),
+    fallback_label=("root", "fallback"),
+    context_storage=dict(),
+    messenger_interface=interface,
+    pre_services=[extract_data],
+)
+
+if __name__ == "__main__":
```

Comment:
See comment to `examples/telegram/interfaces/polling.py`.

Reply:
Replied above